### ISSUE TITLE
Initialising the Android SDK lazily and conditionally

### ISSUE THREAD

**Issue by vibin** [getsentry/sentry-java#2081]
### Description

Hi, i've observed that `sentry-android-core` can contribute anywhere from 5 to 10% in cold startup time overhead.
So i'm evaluating the approach of initialising it on a separate worker thread, so that the Android main thread can
continue starting the rest of the app.

In addition to above, I'm also evaluating introducing a feature flag around SDK initialisation, so that I can
disable it from backend if needed.

Are these two usecases OK or will they pose any problems?

**Comment by marandaneto** [comment]
> So i'm evaluating the approach of initialising it on a separate worker thread, so that the Android main thread can
> continue starting the rest of the app.

Some features require executing on the main thread.
Is the 5 to 10% out of how many milliseconds? our measurement is that the SDK takes around ~20ms to init. only on a mid-end device.

> I'm also evaluating introducing a feature flag around SDK initialisation, so that I can
> disable it from backend if needed.

You can simply call `Sentry.close()` if you want to disable it, or not even init the SDK, what's the use case?

**Comment by vibin** [comment]
@marandaneto 
> Some features require executing on the main thread.

Ah...maybe I can execute it on the main thread, but defer the initialisation via Handler so that it kicks off after Activity frame is drawn? I'll need to handle background process creation case though.

> Is the 5 to 10% out of how many milliseconds? 

On Pixel 4A, I have recorded 20-30ms contribution on P95 cold app open time of 533ms.
On a very low-end device running Android 5.0, I have recorded 250-350ms contrib on P95 cold app open time of 4236ms.

> You can simply call `Sentry.close()` if you want to disable it, or not even init the SDK, what's the use case?

Not initing sounds good. Thanks.
The usecase is trying out sentry-android-core in production without increasing our overall cold app open time metric. (We will be turning on Sentry only for a few test users and do a control vs test comparison in cold startup time)
